plugins/modules/omevv_firmware.py

```diff
@@ -0,0 +1,1224 @@
+            vcenter_uuid, cluster_group_id, new_host_id, parameters['targets'], host_service_tags)
+
+        if self.module.check_mode or self.module.params.get('_ansible_check_mode'):
+            self.handle_check_mode(firmware_update_needed, before_no_change_dict,
+                                   after_no_change_dict, before_dict, after_dict)
+
+        if firmware_update_needed:
+            self.handle_firmware_update(vcenter_uuid, cluster_group_id, payload, parameters,
+                                        before_dict, after_dict)
+
+        else:
+            self.module.exit_json(msg=CHANGES_NOT_FOUND_MSG,
+                                  diff={"before": before_no_change_dict,
+                                        "after": after_no_change_dict}, changed=False)
+
+    def process_cluster_target(self, target):
+        """
+        Process the cluster target.
+
+        Args:
+            target (dict): The target cluster.
+
+        Returns:
+            tuple: A tuple containing the cluster group ID, payload, and new host ID.
+        """
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        host_ids, host_service_tags = self.get_host_id(vcenter_uuid, target)
+        if host_ids is None or not host_ids:
+            self.module.exit_json(msg=CLUSTER_HOST_NOT_FOUND_MSG, skipped=True)
+        cluster_name = target['cluster']
+        cluster_group_id = self.omevv_info_obj.get_group_id_of_cluster(vcenter_uuid,
+                                                                       cluster_name)
+        payload = self.get_payload_details(host_id=host_ids)
+        new_host_id = host_ids
+        return cluster_group_id, payload, new_host_id
+
+    def process_non_cluster_target(self, parameters):
+        """
+        Process the non-cluster target.
+
+        Args:
+            parameters (dict): The parameters dictionary.
+
+        Returns:
+            tuple: A tuple containing the cluster group ID, payload, and new host ID.
+        """
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        target = self.get_target(parameters['targets'])
+        host_id, host_service_tags = self.get_host_from_parameters(vcenter_uuid, parameters)
+        if host_id is None:
+            if target['host']:
+                self.module.exit_json(msg=HOST_NOT_FOUND_MSG.format(
+                    managed_host=target['host']), skipped=True)
+            else:
+                self.module.exit_json(msg=HOST_NOT_FOUND_MSG.format(
+                    managed_host=target['servicetag']), skipped=True)
+        cluster_name = self.omevv_info_obj.get_cluster_name(vcenter_uuid, host_id)
+        cluster_group_id = self.omevv_info_obj.get_group_id_of_cluster(vcenter_uuid,
+                                                                       cluster_name)
+        payload = self.get_payload_details(host_id=host_id)
+        new_host_id = host_id
```

Comment:
new_host_id is only being used in return, remove it

Reply:
Incorporated the comment